rust/lance/src/dataset/optimize.rs

```diff
@@ -968,6 +975,567 @@ pub async fn commit_compaction(
     Ok(metrics)
 }
 
+/// A group of columns that should be stored together
+#[derive(Debug, Clone)]
+pub struct ColumnGroup {
+    pub columns: Vec<String>,
+}
+
+impl ColumnGroup {
+    pub fn new(columns: Vec<String>) -> Self {
+        Self { columns }
+    }
+}
+
+/// Configuration options for Optimize columns.
+///
+/// This struct controls how columns are regrouped within fragments.
+#[derive(Debug, Clone, Default)]
+pub struct OptimizeColumnOptions {
+    /// Specifies how columns should be grouped into files.
+    /// Each [`ColumnGroup`] will result in one data file per fragment.
+    /// If `None`, all columns will be placed in a single file (the default).
+    pub column_groups: Option<Vec<ColumnGroup>>,
```

Comment:
we should just call this something like `optimize_groups`, try to avoid confusion once we introduce ColumnFamily actually as a part of the table metadata manifest.